From 6c51f236ef6bbe463ab6edce17c0c314bdc8ae37 Mon Sep 17 00:00:00 2001
From: kd-11 <karokidii@gmail.com>
Date: Mon, 21 Sep 2026 15:47:31 +0300
Subject: [PATCH 2/5] O2: vk: Fix renderpass keys attached to property blobs
 when seeding interpreter variants

(cherry picked from commit d08d568d5edf4f21508d9b8e471197e6d6f499ad)
---
 rpcs3/Emu/RSX/VK/VKShaderInterpreter.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/rpcs3/Emu/RSX/VK/VKShaderInterpreter.cpp b/rpcs3/Emu/RSX/VK/VKShaderInterpreter.cpp
index 0629cdaef..256428ef9 100644
--- a/rpcs3/Emu/RSX/VK/VKShaderInterpreter.cpp
+++ b/rpcs3/Emu/RSX/VK/VKShaderInterpreter.cpp
@@ -754,8 +754,9 @@ namespace vk
 			pipe_properties.push_back(base_props);
 
 			// Add more targets. Typical G-buffers are 1-4 color targets + depth.
-			for (auto index = 1; index < 4; ++index)
+			for (u8 index = 1; index < 4; ++index)
 			{
+				base_props.renderpass_key = vk::get_renderpass_key(surface_format, depth_format, index + 1);
 				base_props.state.set_attachment_count(index + 1);
 				base_props.state.set_color_mask(index, true, true, true, true);
 				pipe_properties.push_back(base_props);
@@ -766,6 +767,7 @@ namespace vk
 				surface_format != VK_FORMAT_R32_SFLOAT)
 			{
 				// Blending not supported for some F32 formats.
+				base_props.renderpass_key = vk::get_renderpass_key(surface_format, depth_format);
 				base_props.state.set_attachment_count(1);
 				base_props.state.enable_blend(0,
 					VK_BLEND_FACTOR_SRC_ALPHA, VK_BLEND_FACTOR_ONE_MINUS_SRC_ALPHA,
-- 
2.54.0 (Apple Git-157)

